### ISSUE TITLE
linter: typescript/no-meaningless-void-operator doesn't check possibly undefined result

### ISSUE THREAD

**Issue by amarant** [oxc-project/oxc#21373]
### What version of Oxlint are you using?

1.59.0

### What command did you run?

oxlint --config ./.oxlintrc.json

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["typescript"],
  "rules": {
    "typescript/no-meaningless-void-operator": "error",
  }
}
```

### What happened?

A potentially undefined result of a potentially undefined function is not checked against the no-meaningless-void-operator rule :

```typescript
const foo = (() => { }) as (() => void) | undefined;
void foo?.();
```
In typescript-eslint with `"@typescript-eslint/no-meaningless-void-operator": "error"` [it does detect an error](https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=MYewdgzgLgBAZiEMC8MAUaCUKB8MDeMAvtgIYTpa4wBuIAlgCbYA%2BMArmIwKZz1jdGAbgBQdJvEQB%2BAHRZRQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtEBDJswDmSZMloA3DpQAmtDqWh98HaOiiJo0VZHBgAviH1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

but not with oxc (1.59.0) / tsgolint (0.20.0).
